mcp/streamable.go

```diff
@@ -40,12 +41,46 @@ type StreamableHTTPHandler struct {
 	getServer func(*http.Request) *Server
 	opts      StreamableHTTPOptions
 
-	onTransportDeletion func(sessionID string) // for testing only
+	onTransportDeletion func(sessionID string) // for testing
 
-	mu sync.Mutex
-	// TODO: we should store the ServerSession along with the transport, because
-	// we need to cancel keepalive requests when closing the transport.
-	transports map[string]*StreamableServerTransport // keyed by IDs (from Mcp-Session-Id header)
+	mu       sync.Mutex
+	sessions map[string]*sessionInfo // keyed by session ID
+}
+
+type sessionInfo struct {
+	session   *ServerSession
+	transport *StreamableServerTransport
+
+	// If timeout is set, automatically close the session after an idle period.
+	timeout time.Duration
+	timerMu sync.Mutex
+	timer   *time.Timer
+}
+
+// resetTimeout resets the inactivity timer.
+func (i *sessionInfo) resetTimeout() {
+	if i.timeout <= 0 {
+		return
+	}
+
+	i.timerMu.Lock()
+	defer i.timerMu.Unlock()
+
+	if i.timer == nil {
+		return
+	}
+	// Reset the timer if we successfully stopped it.
```

Comment:
"if we successfully stopped it" doesn't appear to be correct; the timer is not stopped. (If it was stopped, it is nil and we don't reset it.)

Reply:
Whoops, yes this was a stale comment from a previous iteration. Thanks for catching.